sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/UDAQuerySuite.scala

```diff
@@ -119,6 +119,15 @@ object CountSerDeAgg extends Aggregator[Int, CountSerDeSQL, CountSerDeSQL] {
   def outputEncoder: Encoder[CountSerDeSQL] = ExpressionEncoder[CountSerDeSQL]()
 }
 
+object ArrayDataAgg extends Aggregator[Array[Double], Double, Double] {
```

Comment:
can we test if the buffer is array type as well?

Reply:
My repl testing uses a buffer that is an array of t-digests, so I assume this is working, but I'll also add a unit test.

Reply:
IIUC this is because the aggregation buffer is using the encoder purely for ser/de, and doesn't need to be resolved with respect to the query plan?

Reply:
actually that's oversimplifying, I am using a UDT containing an array. not the same thing.